src/common/__tests__/hotbar-store.test.ts

```diff
@@ -31,4 +95,137 @@ describe("HotbarStore", () => {
   });
+
+  describe("hotbar items", () => {
+    it("initially creates 12 empty cells", () => {
+      const hotbarStore = HotbarStore.createInstance();
+
+      hotbarStore.load();
+      expect(hotbarStore.getActive().items.length).toEqual(12);
+    });
+
+    it("adds items", () => {
+      const hotbarStore = HotbarStore.createInstance();
+      const entity = new CatalogEntityItem(testCluster);
+
+      hotbarStore.load();
+      hotbarStore.addToHotbar(entity);
+      const items = hotbarStore.getActive().items.filter(Boolean);
+
+      expect(items.length).toEqual(1);
+    });
+
+    it("removes items", () => {
+      const hotbarStore = HotbarStore.createInstance();
+      const entity = new CatalogEntityItem(testCluster);
+
+      hotbarStore.load();
+      hotbarStore.addToHotbar(entity);
+      hotbarStore.removeFromHotbar("test");
+      const items = hotbarStore.getActive().items.filter(Boolean);
+
+      expect(items.length).toEqual(0);
+    });
+
+    it("does nothing if removing with invalid uid", () => {
+      const hotbarStore = HotbarStore.createInstance();
+      const entity = new CatalogEntityItem(testCluster);
+
+      hotbarStore.load();
+      hotbarStore.addToHotbar(entity);
+      hotbarStore.removeFromHotbar("invalid uid");
+      const items = hotbarStore.getActive().items.filter(Boolean);
+
+      expect(items.length).toEqual(1);
+    });
+
+    it("moves item to empty cell", () => {
+      const hotbarStore = HotbarStore.createInstance();
+      const test = new CatalogEntityItem(testCluster);
+      const minikube = new CatalogEntityItem(minikubeCluster);
+      const aws = new CatalogEntityItem(awsCluster);
+
+      hotbarStore.load();
+      hotbarStore.addToHotbar(test);
+      hotbarStore.addToHotbar(minikube);
+      hotbarStore.addToHotbar(aws);
+
+      expect(hotbarStore.getActive().items[5]).toBeNull();
+
+      hotbarStore.restackItems(1, 5);
+
+      expect(hotbarStore.getActive().items[5]).toBeTruthy();
+      expect(hotbarStore.getActive().items[5].entity.uid).toEqual("minikube");
+    });
+
+    it("moves items down", () => {
+      const hotbarStore = HotbarStore.createInstance();
+      const test = new CatalogEntityItem(testCluster);
+      const minikube = new CatalogEntityItem(minikubeCluster);
+      const aws = new CatalogEntityItem(awsCluster);
+
+      hotbarStore.load();
+      hotbarStore.addToHotbar(test);
+      hotbarStore.addToHotbar(minikube);
+      hotbarStore.addToHotbar(aws);
+
+      // aws -> test
+      hotbarStore.restackItems(2, 0);
+
+      const items = hotbarStore.getActive().items.map(item => item?.entity.uid || null);
+
+      expect(items.slice(0, 4)).toEqual(["aws", "test", "minikube", null]);
+    });
+
+    it("moves items up", () => {
+      const hotbarStore = HotbarStore.createInstance();
+      const test = new CatalogEntityItem(testCluster);
+      const minikube = new CatalogEntityItem(minikubeCluster);
+      const aws = new CatalogEntityItem(awsCluster);
```

Comment:
Why do we need such tests? 
Wouldn't be better to check behaviour of `HotbarStore.restackItems()`in dedicated test?

Reply:
What you mean by dedicated tests? Introduce separate file for dragndrop behaviour or test it within ui tests?

Reply:
I meant instead of having tests like "_moves items **up**/**down/etc**_" better to create separated test-case:

```ts
test(("Hotbar.restackItems() allows to reorganize hotbar items") => {
 // test method logic and just check items-list in the end
})
```

Because in the end you're checking logic for `restackItems()` after some manipulations with involving other methods like `addToHotbar()` (which is against unit-testing nature). Wdyt?